File: app-miyou/src/main/java/com/miyou/app/domain/memory/port/ConversationCounterPort.java
package com.miyou.app.domain.memory.port;

import com.miyou.app.domain.dialogue.model.ConversationSessionId;
import reactor.core.publisher.Mono;

public interface ConversationCounterPort {

	Mono<Long> increment(ConversationSessionId sessionId);

	Mono<Long> get(ConversationSessionId sessionId);

	Mono<Void> reset(ConversationSessionId sessionId);
}

File: app-miyou/src/main/kotlin/com/miyou/app/domain/memory/port/ConversationCounterPort.kt
package com.miyou.app.domain.memory.port

import reactor.core.publisher.Mono

interface ConversationCounterPort {
    fun increment(sessionId: String): Mono<Long>

    fun get(sessionId: String): Mono<Long>

    fun reset(sessionId: String): Mono<Void>
}
